test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml

```diff
@@ -296,6 +296,24 @@
       that:
         - azure_vm.properties.availabilitySet.id
 
+- name: Retrieve vms facts (filtering by name)
```

Comment:
I'm not going to hold up the PR for this but it would be good if you could do a test without `name` at some point in the future.